paddle/fluid/memory/allocation/allocator_facade.cc

```diff
@@ -70,7 +70,7 @@ class AllocatorFacadePrivate {
           InitNaiveBestFitXPUAllocator(platform::XPUPlace(dev_id));
         }
 #endif
-#ifdef PADDLE_WITH_CUDA
+#if defined(PADDLE_WITH_CUDA) || defined(PADDLE_WITH_HIP)
```

Comment:
Maybe we can create a new MACRO, eg,
`PADDLE_WITH_GPU  = PADDLE_WITH_CUDA|| PADDLE_WITH_HIP`

Reply:
Keep PADDLE_WITH_HIP as cudnn vs miopen has big difference.